mos-platform/geos-cbm/geos_crt.c

```diff
@@ -0,0 +1,2013 @@
+/***********************************************************************
+ * geos_crt.c
+ *
+ * This file contains the implemantation of the "C" definition of
+ * all geos OS calls for use in GEOS applications  with llvm-mos
+ * C compiler.
+ *
+ * Transcribed from geoProgrammer manual by DiSc.
+ *
+ ***********************************************************************/
+
+#include "geos.h"
+
+/***********************************************************************
+ * GEOS Graphic routines
+ ***********************************************************************/
+
+void DrawPoint(uint16_t x, uint8_t y, bool recover, bool background)
+{
+  __r3 = x;
+  __r11L = y;
+  if (recover) {
+    asm volatile("lda #0xff \n");
```

Comment:
Not entirely sure if it's needed, but the wiki suggest that for performance reasons, inline asm should be marked with `__attribute__((leaf))`. See https://llvm-mos.org/wiki/Optimization_guide.

Reply:
Replaced all occurences